### ISSUE TITLE
Attempt to unlock mutex that was not locked

### ISSUE THREAD

**Issue by Geyup** [Alexey-Yakovenko/deadbeef#1231]
I'm using Slackware 14.1 x86_64 and it comes with glib2-2.36.4 which works fine for deadbeef.

But I recently upgraded glib2 to this version, ```glib2-2.42.2```, and I then noticed that deadebeef did not want to play any longer, it would give me this message at startup;

```Attempt to unlock mutex that was not locked```

I then found this patch you created that worked on this bug report;

https://bugs.archlinux.org/task/42382

https://github.com/Alexey-Yakovenko/deadbeef/commit/46a4f32719ba502e6a20c65a0e8009e67886eaec.patch

So I wanted to report for deadbeef 0.6.2 unless the patch is used deadbeef won't play, not sure how important this is, and if fixed now in a commit, or future commit.

Also when beginning the compile I noticed this information I also wanted to share, I'm  not sure if there is any concern with the infomation it shows below;

```
checking for a BSD-compatible install... /usr/bin/ginstall -c
checking whether build environment is sane... yes
/tmp/foo/deadbeef-0.6.2/missing: Unknown `--is-lightweight' option
Try `/tmp/foo/deadbeef-0.6.2/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
```

**Comment by Alexey-Yakovenko** [comment]
Hi.

> So I wanted to report for deadbeef 0.6.2 unless the patch is used deadbeef won't play, not sure how important this is, and if fixed now in a commit, or future commit.

Obviously, the problem appeared after 0.6.2, and the patch was released after 0.6.2. You should build from git branch "0.6.2-fixes", or use the official binary builds from website.

> I'm not sure if there is any concern with the infomation it shows below

unless it breaks anything -- there are no concerns :)

**Comment by Geyup** [comment]
Thank you I will grab the latest from git.

Keep up the great work, I really love deadbeef! :)

Cheers
